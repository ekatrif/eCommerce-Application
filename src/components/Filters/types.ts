import { IBrand } from './Checkbox/BrandCheckbox/types';
import { IColor } from './Checkbox/ColorCheckbox/types';
import { ISize } from './Checkbox/SizeCheckbox/types';

export type IFilterItem = IBrand['brand'] | IColor['color'] | ISize['size'];
