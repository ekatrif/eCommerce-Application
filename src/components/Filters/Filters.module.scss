.container {
  display: flex;
  flex-direction: column;
  gap: 1em;
  margin: 1em 0;
}

.filter {
  position: relative;
  display: flex;
  flex-direction: column;
  gap: 0.5em;
  margin-top: 2em;

  &:not(:last-of-type):after {
    position: absolute;
    bottom: -1em;
    left: 0;
    content: '';
    width: 100%;
    border-bottom: 1px solid #E5E5E5;
    ;
  }
}

.title {
  margin-bottom: 0.5em;
}

@media screen and (max-width: 400px) {
  .container {
    max-width: 10vw;
  }
}