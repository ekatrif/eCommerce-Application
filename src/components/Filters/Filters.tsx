import React from 'react';
import BrandCheckbox from './Checkbox/BrandCheckbox/BrandCheckbox';
import ColorCheckbox from './Checkbox/ColorCheckbox/ColorCheckbox';
import PriceSlider from './PriceSlider/PriceSlider';
import SizeCheckbox from './Checkbox/SizeCheckbox/SizeCheckbox';
import classes from './Filters.module.scss';
import { IFilters } from './types';
import { IPrice } from './PriceSlider/types';
import Button from '../UI/button/Button';

const Filters: React.FC<IFilters> = ({
  brands,
  colors,
  sizes,
  handleChangeCheckedBrand,
  handleChangeCheckedColor,
  handleChangeCheckedSize,
  prices,
  selectedPrice,
  changePrice,
}) => {
  return (
    <div className={classes.container}>
      <div className={classes.filter}>
        <h4 className={classes.title}>Brands</h4>
        {brands.map((brand, index) => (
          <BrandCheckbox key={index} brand={brand} handleChangeCheckedBrand={handleChangeCheckedBrand} />
        ))}
      </div>
      <div className={classes.filter}>
        <h4 className={classes.title}>Colors</h4>
        {colors.map((color, index) => (
          <ColorCheckbox key={index} color={color} handleChangeCheckedColor={handleChangeCheckedColor} />
        ))}
      </div>
      <div className={classes.filter}>
        <h4 className={classes.title}>Sizes</h4>
        {sizes.map((size, index) => (
          <SizeCheckbox key={index} size={size} handleChangeCheckedSize={handleChangeCheckedSize} />
        ))}
      </div>
      <div className={classes.filter}>
        <h4 className={classes.title}>Price</h4>
        <PriceSlider
          price={{ min: prices[0], max: prices[prices.length - 1] } as IPrice['price']}
          value={selectedPrice}
          changePrice={changePrice || []}
        />
      </div>
    </div>
  );
};

export default Filters;
