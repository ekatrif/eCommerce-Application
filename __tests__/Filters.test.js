import React from 'react';
import renderer from 'react-test-renderer';
import Filters from '../src/components/Filters/Filters'

test('Filters renders correctly', () => {
  const tree = renderer
    .create(<Filters
      brands={['A', 'B']}
      handleChangeCheckedBrand=""
      colors={['A', 'B']}
      handleChangeCheckedColor=""
      sizes={['A', 'B']}
      handleChangeCheckedSize=""
      prices={['A', 'B']}
      selectedPrice="50"
      changePrice=""
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});