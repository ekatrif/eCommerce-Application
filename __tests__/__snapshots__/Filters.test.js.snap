// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="filter"
  >
    <h4
      className="title"
    >
      Brands
    </h4>
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Terranova
        </span>
      </label>
    </div>
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          SomeOther
        </span>
      </label>
    </div>
  </div>
  <div
    className="filter"
  >
    <h4
      className="title"
    >
      Colors
    </h4>
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          white
        </span>
      </label>
    </div>
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          gray
        </span>
      </label>
    </div>
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          blue
        </span>
      </label>
    </div>
  </div>
  <div
    className="filter"
  >
    <h4
      className="title"
    >
      Price
    </h4>
    <div
      className="slider"
    >
      <span
        className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-u259bk-MuiSlider-root"
        onMouseDown={[Function]}
      >
        <span
          className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
        />
        <span
          className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style={
            {
              "left": "0%",
              "width": "30%",
            }
          }
        />
        <span
          className="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
          data-index={0}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            {
              "left": "30%",
              "pointerEvents": undefined,
            }
          }
        >
          <input
            aria-label="Volume"
            aria-orientation="horizontal"
            aria-valuemax={100}
            aria-valuemin={0}
            aria-valuenow={30}
            data-index={0}
            disabled={false}
            max={100}
            min={0}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            step={1}
            style={
              {
                "border": 0,
                "clip": "rect(0 0 0 0)",
                "direction": "ltr",
                "height": "100%",
                "margin": -1,
                "overflow": "hidden",
                "padding": 0,
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "100%",
              }
            }
            type="range"
            value={30}
          />
        </span>
      </span>
    </div>
  </div>
</div>
`;
